app/models/host/discovered.rb

```diff
@@ -133,26 +133,39 @@ def proxied?
     subnet.present? && subnet.discovery.present?
   end
 
+  def ip4or6
+    if Setting[:discovery_prefer_ipv6]
+      IPAddr.new(self.ip6 || self.ip)
+    else
+      IPAddr.new(self.ip || self.ip6)
+    end
+  end
+
   def proxy_url(node_ip)
-    proxied? ? subnet.discovery.url + "/discovery/#{node_ip}" : "https://#{node_ip}:8443"
+    wrapped_ip = node_ip.ipv6? ? "[#{node_ip}]" : node_ip
```

Comment:
While looking at this again, I believe there must be a better way (my idea was `IPAddr`) to wrap the IP ourselves.

Reply:
So you don't like the if statement and want to figure out something better? Problem is, URL and IPAddr do not play nice together. There is no IPAddr.hostname field that would return either wrapped or unwrapped version.

```
>> URI::HTTP.build(host: IPAddr.new("1.2.3.4"))
Traceback (most recent call last):
       12: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `<main>'
       11: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `load'
       10: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        9: from (irb):6
        8: from (irb):7:in `rescue in irb_binding'
        7: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/http.rb:62:in `build'
        6: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/generic.rb:137:in `build'
        5: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/generic.rb:137:in `new'
        4: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/generic.rb:191:in `initialize'
        3: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/generic.rb:663:in `hostname='
        2: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/generic.rb:663:in `!~'
        1: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/generic.rb:663:in `=~'
TypeError (no implicit conversion of IPAddr into String)
```

Reply:
Ok, I guess this was just me being nit-picky.

Reply:
You are right and it is ugly code, well, I did my homework and research this. There is nothing in both IPAdd and URI classes (common, generic, HTTP(s)) that would make this easier. :-(